recipes/ulfius/all/patches/2.7.11-0001-shared-static-conan.patch

```diff
@@ -0,0 +1,895 @@
+@@ -459,12 +518,35 @@ if (INSTALL_HEADER)
+     install(FILES API.md
+             DESTINATION ${CMAKE_INSTALL_DOCDIR} COMPONENT runtime)
+ else ()
+-install(TARGETS ${TARGETS}
++install(TARGETS ${TARGETS} EXPORT UlfiusExports
+         RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}
+         LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
+         ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR})
+ endif ()
+ 
++set(ULFIUS_INSTALL_CMAKEDIR_DEFAULT "${CMAKE_INSTALL_LIBDIR}/cmake/Ulfius")
++if (WIN32 AND NOT MINGW)
++    set(ULFIUS_INSTALL_CMAKEDIR_DEFAULT "cmake")
++endif ()
++set(ULFIUS_INSTALL_CMAKEDIR ${ULFIUS_INSTALL_CMAKEDIR_DEFAULT} CACHE STRING "Location where to install the cmake config files")
++
++install(EXPORT UlfiusExports DESTINATION "${ULFIUS_INSTALL_CMAKEDIR}"
++    NAMESPACE "Ulfius::"
++    FILE "UlfiusTargets.cmake")
++
++configure_package_config_file(cmake-modules/UlfiusConfig.cmake.in UlfiusConfig.cmake
++    INSTALL_DESTINATION "${ULFIUS_INSTALL_CMAKEDIR}")
++write_basic_package_version_file(UlfiusConfigVersion.cmake
++    COMPATIBILITY AnyNewerVersion)
++
++install(FILES
++            cmake-modules/FindGnuTLS.cmake
++            cmake-modules/FindJansson.cmake
++            cmake-modules/FindMHD.cmake
++            "${PROJECT_BINARY_DIR}/UlfiusConfig.cmake"
++            "${PROJECT_BINARY_DIR}/UlfiusConfigVersion.cmake"
++        DESTINATION "${ULFIUS_INSTALL_CMAKEDIR}")
++
+ # uninstall target
+ 
+ if (NOT TARGET uninstall)
+@@ -563,6 +645,7 @@ message(STATUS "Outgoing requests support: ${WITH_CURL}")
+ message(STATUS "Jansson library support:   ${WITH_JANSSON}")
+ message(STATUS "Yder support:              ${WITH_YDER}")
+ message(STATUS "Build uwsc application:    ${BUILD_UWSC}")
++message(STATUS "Build shared library:      ${BUILD_SHARED}")
+ message(STATUS "Build static library:      ${BUILD_STATIC}")
+ message(STATUS "Build testing tree:        ${BUILD_ULFIUS_TESTING}")
+ message(STATUS "Install the header files:  ${INSTALL_HEADER}")
+diff --git a/cmake-modules/FindCheck.cmake b/cmake-modules/FindCheck.cmake
+index 4aad6bc..30221fc 100644
+--- a/cmake-modules/FindCheck.cmake
++++ b/cmake-modules/FindCheck.cmake
+@@ -68,6 +68,12 @@ find_package_handle_standard_args(Check
+ if (CHECK_FOUND)
+     set(CHECK_LIBRARIES ${CHECK_LIBRARY})
+     set(CHECK_INCLUDE_DIRS ${CHECK_INCLUDE_DIR})
++    if (NOT TARGET Check::Check)
++        add_library(Check::Check UNKNOWN IMPORTED)
++        set_target_properties(Check::Check PROPERTIES
++                IMPORTED_LOCATION "${CHECK_LIBRARY}"
++                INTERFACE_INCLUDE_DIRECTORIES "${CHECK_INCLUDE_DIR}")
++    endif ()
+ endif ()
+ 
+-mark_as_advanced(CHECK_INCLUDE_DIR CHECK_LIBRARY)
+\ No newline at end of file
++mark_as_advanced(CHECK_INCLUDE_DIR CHECK_LIBRARY)
+diff --git a/cmake-modules/FindGnuTLS.cmake b/cmake-modules/FindGnuTLS.cmake
+new file mode 100644
+index 0000000..aa97c86
+--- /dev/null
++++ b/cmake-modules/FindGnuTLS.cmake
```

Comment:
Shouldn't we be using the Conan one (when it's added) instead of patches?

Reply:
I removed these patches.
 gnutls is not available yet, but when it becomes available, `generate()` should make sure to create `GnuTLSConfig.cmake` and create the `GnuTLS::GnuTLS` target.